index.js

```diff
@@ -51,22 +50,30 @@ function licensee (configuration, path, callback) {
       // dependency tree.  Calling read-package-tree twice, at
       // the same time, is far from efficient.  But it works,
       // and doing so helps keep this package small.
-      runParallel({
-        dependencies: readDependencyList,
-        packages: readFilesystemTree
-      }, function (error, trees) {
-        if (error) callback(error)
-        else withTrees(trees.packages, trees.dependencies)
-      })
+      let packages, dependencies
+      try {
+        [packages, dependencies] = await Promise.all([
+          readFilesystemTree(),
+          readDependencyList()
+        ])
+      } catch (error) {
+        callback(error)
```

Comment:
Would it not make more sense to let it throw then?

Reply:
Yes. But since there would indeed be a little more work to support both callback and Promise returning, I was trying to honor @kemitchell 's [preference](https://github.com/jslicense/licensee.js/pull/55#issuecomment-558437231) for not adding complexity in supporting both callback and Promise; I thus confined myself here to those *internal* Promise additions which were strictly necessary for removing the dependency.

But if y'all are open to a breaking change, then yes, a public Promise API would make this change cleaner as well (and of course easier for users who wouldn't need to clutter their code with extra `promisify` statements).

Reply:
My bad. To clarify, there is no "y'all" here :) This is really @kemitchell's project, and I'm an interested bystander. Let me give this review another shot.

Reply:
There's still the problem that errors will not find their way to the promise.

How about keeping the function as-is, but wrapping it with a function that takes care of the promisification?

Something like this:

```js
function licensee(configuration, path, callback) {
  if (arguments.length < 3) {
    return new Promise(function (resolve, reject) {
      realLicensee(configuration, path, function (error, result) {
        if (error) {
          reject(error);
        } else {
          resolve(result);
        }
      });
    });
  } else {
    realLicensee(configuration, path, callback);
  }
}
```

Then you have both patterns without having to change the `licensee` logic at all.

Reply:
I'm indeed happier with that and would have done that, as I'd like to see a built-in Promise if that's ok with @kemitchell , but I've gotten the impression this whole PR may only end up merged in some form if it's needed by `read-package-tree` per https://github.com/jslicense/licensee.js/pull/62#issuecomment-558922794 and even then, it sounds like, per https://github.com/jslicense/licensee.js/pull/62#issuecomment-558915549 , that  `promisify` is going to be expected rather than having the project providing its own Promise-returning ability . The way I implemented still works in a backward compatible way even if it doesn't fully take advantage of Promises so that the public API allows it (including sending Promise users rejections upon errors) as in your sample.

Reply:
This PR is indeed on hold. See #64.